lib/cartopy/mpl/geoaxes.py

```diff
@@ -392,6 +392,16 @@ def hold_limits(self, hold=True):
                 (self.ignore_existing_data_limits,
                     self._autoscaleXon, self._autoscaleYon) = other
 
+    def get_tightbbox(self, *args, **kwargs):
+        """Draw gridliner objects to tight bounding box algorithm will
+        incorporate gridliner labels."""
+        # Matplotlib approaches this problem in the same way, by duplicating
+        # post-processing steps in both Axes.draw and Axes.get_tightbbox
+        for gl in self._gridliners:
+            gl._draw_gridliner(background_patch=self.background_patch)
+        self._gridliners = []
```

Comment:
I may not understand the rest of CartoPy well enough, but why are you resetting `self._gridliners`?

Reply:
This is because `draw` and `get_tight_bbox` can be called more than once (for example, if the user calls `Figure.show` more than once, or if an interactive backend automatically updates figure changes). Without `self._gridliners = []` they would be drawn multiple times. Note this code block is copied directly from `GeoAxes.draw`.